import.sh

```diff
@@ -2,195 +2,29 @@
-src/google/protobuf/compiler/java/java_message.cc
-src/google/protobuf/compiler/java/java_message_builder.cc
-src/google/protobuf/compiler/java/java_message_builder_lite.cc
-src/google/protobuf/compiler/java/java_message_field.cc
-src/google/protobuf/compiler/java/java_message_field_lite.cc
-src/google/protobuf/compiler/java/java_message_lite.cc
-src/google/protobuf/compiler/java/java_name_resolver.cc
-src/google/protobuf/compiler/java/java_primitive_field.cc
-src/google/protobuf/compiler/java/java_primitive_field_lite.cc
-src/google/protobuf/compiler/java/java_service.cc
-src/google/protobuf/compiler/java/java_shared_code_generator.cc
-src/google/protobuf/compiler/java/java_string_field.cc
-src/google/protobuf/compiler/java/java_string_field_lite.cc
-src/google/protobuf/compiler/javanano/javanano_enum.cc
-src/google/protobuf/compiler/javanano/javanano_enum_field.cc
-src/google/protobuf/compiler/javanano/javanano_extension.cc
-src/google/protobuf/compiler/javanano/javanano_field.cc
-src/google/protobuf/compiler/javanano/javanano_file.cc
-src/google/protobuf/compiler/javanano/javanano_generator.cc
-src/google/protobuf/compiler/javanano/javanano_helpers.cc
-src/google/protobuf/compiler/javanano/javanano_map_field.cc
-src/google/protobuf/compiler/javanano/javanano_message.cc
-src/google/protobuf/compiler/javanano/javanano_message_field.cc
-src/google/protobuf/compiler/javanano/javanano_primitive_field.cc
-src/google/protobuf/compiler/js/js_generator.cc
-src/google/protobuf/compiler/objectivec/objectivec_enum.cc
-src/google/protobuf/compiler/objectivec/objectivec_enum_field.cc
-src/google/protobuf/compiler/objectivec/objectivec_extension.cc
-src/google/protobuf/compiler/objectivec/objectivec_field.cc
-src/google/protobuf/compiler/objectivec/objectivec_file.cc
-src/google/protobuf/compiler/objectivec/objectivec_generator.cc
-src/google/protobuf/compiler/objectivec/objectivec_helpers.cc
-src/google/protobuf/compiler/objectivec/objectivec_map_field.cc
-src/google/protobuf/compiler/objectivec/objectivec_message.cc
-src/google/protobuf/compiler/objectivec/objectivec_message_field.cc
-src/google/protobuf/compiler/objectivec/objectivec_oneof.cc
-src/google/protobuf/compiler/objectivec/objectivec_primitive_field.cc
-src/google/protobuf/compiler/plugin.cc
-src/google/protobuf/compiler/plugin.pb.cc
-src/google/protobuf/compiler/python/python_generator.cc
-src/google/protobuf/compiler/ruby/ruby_generator.cc
-src/google/protobuf/compiler/subprocess.cc
-src/google/protobuf/compiler/zip_writer.cc
-'
+# symlink so cgo compiles them
+for file in $LIBPROTOBUF_LITE_SOURCES $LIBPROTOBUF_SOURCES; do
+  ln -sf internal/src/$file .
+done
 
 # symlink so cgo compiles them
-for file in $PROTOC_SOURCES; do
-  ln -sf ../internal/$file libprotoc
+for file in $LIBPROTOC_SOURCES; do
+  ln -sf ../../internal/src/$file cmd/protoc
 done
+
+# restore the repo to what it would look like when first cloned.
+# comment this line out while updating upstream.
+git clean -dxf
```

Comment:
Yeah, maybe, but can we do that in the future? We'll have to apply that
treatment to all the c-* repos and I don't feel like working in this
gigantic diff any longer than necessary.

On Wed, May 25, 2016 at 7:23 PM, Ben Darnell <notifications@github.com>
wrote:

> In import.sh
> <https://github.com/cockroachdb/c-protobuf/pull/12#discussion_r64670098>:
>
> >  done
> > +
> > +# restore the repo to what it would look like when first cloned.
> > +# comment this line out while updating upstream.
> > +git clean -dxf
>
> It sounds like it would be better to put some instructions somewhere that
> say "run import.sh, git clean -fdx, then run import.sh again" instead of
> having import.sh do the clean on its own and require commenting out. On
> Wed, May 25, 2016 at 7:21 PM, Tamir Duberstein <notifications@github.com>
> wrote:
> … <#m_-2343822673542274620_>
> In import.sh <#12 (comment)
> <https://github.com/cockroachdb/c-protobuf/pull/12#discussion_r64669906>>:
> > done > + > +# restore the repo to what it would look like when first
> cloned. > +# comment this line out while updating upstream. > +git clean
> -dxf You'd run it when updating upstream, then you'd commit all your stuff,
> and then you'd run it once more to make sure your build didn't depend on
> something you had gitignored. … <#m_-2626471576060581338_> On Wed, May 25,
> 2016 at 7:18 PM, Ben Darnell ***@***.***> wrote: In import.sh <#12
> <https://github.com/cockroachdb/c-protobuf/pull/12> (comment) <#12
> (comment)
> <https://github.com/cockroachdb/c-protobuf/pull/12#discussion_r64669506>>>:
> > done > + > +# restore the repo to what it would look like when first
> cloned. > +# comment this line out while updating upstream. > +git clean
> -dxf Isn't updating upstream the common case for running import.sh? When
> would you run this other than when you're updating upstream? — You are
> receiving this because you authored the thread. Reply to this email
> directly or view it on GitHub <
> https://github.com/cockroachdb/c-protobuf/pull/12/files/74267fc321d5177b8ccb3a0dfb80ab3a463f2c87#r64669506>
> — You are receiving this because you commented. Reply to this email
> directly or view it on GitHub <
> https://github.com/cockroachdb/c-protobuf/pull/12/files/74267fc321d5177b8ccb3a0dfb80ab3a463f2c87#r64669906>
>
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/c-protobuf/pull/12/files/74267fc321d5177b8ccb3a0dfb80ab3a463f2c87#r64670098>
>